src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/TestRouter.java

```diff
@@ -0,0 +1,172 @@
+ * limitations under the License.
+ */
+
+package com.feedzai.commons.sql.abstraction.engine.impl.abs;
+
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+import static com.feedzai.commons.sql.abstraction.engine.configuration.PdbProperties.JDBC;
+
+/**
+ * A class that acts as a router, to forward data sent between a {@link DatabaseEngine} and the DB.
+ * <p>
+ * When running tests, this offers the possibility of interrupting connections without closing them.
+ *
+ * @author José Fidalgo (jose.fidalgo@feedzai.com)
+ */
+public class TestRouter implements Closeable {
+
+    /**
+     * Logger for this class.
+     */
+    private static final Logger logger = LoggerFactory.getLogger(TestRouter.class);
+
+    /**
+     * The {@link ServerSocket} that will accept connections from PDB engines to this {@link TestRouter}.
+     */
+    private final ServerSocket serverSocket = new ServerSocket(0);
+
+    /**
+     * The {@link Socket} to connect this {@link TestRouter} to the database.
+     */
+    private final Socket socketRouterToDb = new Socket();
+
+    /**
+     * An {@link ExecutorService} to handle asynchronous tasks.
+     */
+    private final ExecutorService executor;
+
+    /**
+     * The database port.
+     */
+    private final int dbPort;
+
+    /**
+     * Whether this {@link TestRouter} is running.
+     */
+    private volatile boolean isRunning = true;
+
+    /**
+     * Constructor for this class.
+     *
+     * @param executor An {@link ExecutorService} to run tasks asynchronously.
+     * @param dbPort   The real DB server port.
+     * @throws IOException if some problem occurs creating a new {@link ServerSocket}.
+     */
+    TestRouter(final ExecutorService executor, final int dbPort) throws IOException {
+        this.executor = executor;
+        this.dbPort = dbPort;
+    }
+
+    /**
+     * Gets new DB properties with the JDBC URL patched to replace the default DB port with a new one.
+     *
+     * @param dbProperties The original DB properties.
+     * @param defaultPort  The default DB port.
+     * @param newPort      The port to use as replacement in the JDBC URL for the default DB port.
+     * @return The patched DB properties.
+     */
+    public static Properties getPatchedDbProperties(final Properties dbProperties, final int defaultPort, final int newPort) {
+        final Properties testProps = new Properties();
+        testProps.putAll(dbProperties);
+
+        final String patchedJdbc = dbProperties.getProperty(JDBC)
+                .replace("localhost:" + defaultPort, "localhost:" + newPort);
```

Comment:
This will just replace the port if the hostname is `localhost`. IMO either this just replace the port and change the javadocs  to say that it only replaces the port if the hostname is `localhost`